client/src/reducers/controls.js

```diff
@@ -75,6 +78,39 @@ const Controls = (
       const categoricalAsBooleansMap = createCategoricalAsBooleansMap(world);
       const crossfilter = Crossfilter(world.obsAnnotations);
       const dimensionMap = World.createObsDimensionMap(crossfilter, world);
+
+      let worldVarDataCache = world.varDataCache;
+
+      /* var dimensions */
+      if (userDefinedGenes.length > 0) {
+        /*
+          verbose & slightly confusing that we also access this as an object
+          in controls rather than an array, should be abstracted into
+          util ie., createDimensionsFromBothListsOfGenes(userGenes, diffExp)
+        */
+        _.forEach(userDefinedGenes, (gene, index) => {
+          dimensionMap[gene] = World.createVarDimension(
+            /* "__var__" + */
+            world,
+            worldVarDataCache,
+            crossfilter,
+            gene
+          );
+        });
+      }
+
+      if (diffexpGenes.length > 0) {
```

Comment:
so what happens if the same gene is in both userDefinedGenes and diffexpGenes?  Do you really want two dimensions created?  Currently, the first one is leaked (memory leak because dispose is never called when it is over-written).

I suspect you want to create a single (unique) dimension for each 'gene', so you probably should it on the union of userDefinedGenes and diffexpGenes